File: mcp-mongo-project/src/utils/connection.js
import pkg from 'pg';
const { Pool } = pkg;

export class DatabaseConnection {
    constructor() {
        this.pool = null;
    }

    static getInstance() {
        if (!DatabaseConnection.instance) {
            DatabaseConnection.instance = new DatabaseConnection();
        }
        return DatabaseConnection.instance;
    }

    async connect(connectionString) {
        if (this.pool) {
            await this.disconnect();
        }

        this.pool = new Pool({
            connectionString,
            ssl: process.env.NODE_ENV === 'production' ? { rejectUnauthorized: false } : false
        });

        // Проверяем соединение
        const client = await this.pool.connect();
        client.release();

        console.log('Connected to PostgreSQL');
    }

    async disconnect() {
        if (this.pool) {
            await this.pool.end();
            this.pool = null;
            console.log('Disconnected from PostgreSQL');
        }
    }

    async query(text, params = []) {
        if (!this.pool) {
            throw new Error('Database not connected');
        }

        const result = await this.pool.query(text, params);
        return result.rows;
    }
}

File: equipmetry/mcp_mongo_postgres/utils/connection.py
import asyncpg
import os

class DatabaseConnection:
    instance = None
    
    def __init__(self):
        self.pool = None

    @staticmethod
    def get_instance():
        if not DatabaseConnection.instance:
            DatabaseConnection.instance = DatabaseConnection()
        return DatabaseConnection.instance

    async def connect(self, connection_string):
        if self.pool:
            await self.disconnect()

        # Configure SSL based on environment
        if os.environ.get('NODE_ENV') == 'production':
            ssl_config = 'require'  # Equivalent to { rejectUnauthorized: false }
        else:
            ssl_config = False

        self.pool = await asyncpg.create_pool(
            connection_string,
            ssl=ssl_config
        )

        # Проверяем соединение
        async with self.pool.acquire() as client:
            pass  # Connection acquired and will be released automatically

        print('Connected to PostgreSQL')

    async def disconnect(self):
        if self.pool:
            await self.pool.close()
            self.pool = None
            print('Disconnected from PostgreSQL')

    async def query(self, text, params=None):
        if params is None:
            params = []
            
        if not self.pool:
            raise Exception('Database not connected')

        async with self.pool.acquire() as connection:
            result = await connection.fetch(text, *params)
            # Convert asyncpg Records to dictionaries to match pg.js behavior
            return [dict(row) for row in result]
